### PR TITLE
Node module exports.

### PR DESCRIPTION
**Pull request by jjzazuet**
OK let's try this one more time.

These changes add `node` style `require()` exports to `nvd3`, along with package information for the npm registry. 

Specifically, this allows browserify/webpack users to require `nvd3`, along with its peer dependency, `d3` inside a Browser/CommonJS environment. It works for me and my application. Let me know if you want a unit test.

Thanks.

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -140,4 +139,13 @@ nv.addGraph = function(obj) {
     if (!nv.render.active) {
         nv.render();
     }
-};
\ No newline at end of file
+};
+
+// exports
```
Comment (robinfhu) [inline src/core.js]: Change this comment to say something related to node.js

**Comment by robinfhu** [comment]
And yes can you add a unit test?  Maybe the test involves creating a window.module object and checking if module.exports=nv

**Comment by jjzazuet** [comment]
OK done. I've included a small project inside the `test/node` folder. It basically renders a chart using CommonJS style exports. To verify this:

Build `nvd3` as usual. Change to `test/node`. Install the test dependencies (needs `gulp`). Start an http server and browse to the example page.

    nvd3 $ grunt production
    nvd3 $ cd test/node
    nvd3/test/node $ npm install .
    nvd3/test/node $ gulp
    nvd3/test/node $ python -m SimpleHTTPServer 8000

Browse to `http://localhost:8000/nodeTest.html`

Thanks!

**Comment by robinfhu** [comment]
Hi, why does it need gulp, which is just another program we will need?  Can this be done in grunt?

File: package.json [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
   },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/novus/nvd3"
+  },
+  "keywords": ["nvd3", "d3", "chart", "graph"],
+  "readmeFilename": "README.md",
+  "license": "Apache-2.0",
+  "peerDependencies": {
+    "d3": "<=3.3.13"
```
Comment (liquidpele) [inline package.json]: We just increased the required d3 to 3.4.4

**Comment by robinfhu** [comment]
In any case, I can't test this on my Windows machine (npm install on 'contextify' is failing), so I'll wait till I have my Mac.   I do think this is intriguing and would like to see it in action.  Also, can you put a  README in the test/node folder with the steps you described.

**Comment by jjzazuet** [comment]
Ok done.

- D3 version has been bumped to `3.4.4`.
- You now have a grunt file.

Not to start a religious war but, I kinda like `gulp`'s stream based mechanism a tad better than `grunt`. But if that's how you roll well... yeah, you got it.

Thanks!

**Comment by robinfhu** [comment]
@jjzazuet yeah, I just realized that it doesn't really matter, but it would keep things consistent at least.

**Comment by jjzazuet** [comment]
Done. Added `README.md`. Thanks!

**Comment by robinfhu** [comment]
Thanks. Give me till tomorrow to merge this in.

**Pull request merged**